Vagrantfile

```diff
@@ -0,0 +1,80 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+# Original version of this file copied from: https://raw.githubusercontent.com/mitchellh/packer/master/Vagrantfile
+#
+
+GOROOT = '/opt/go'
+GOPATH = '/opt/gopath'
+PACKAGE_PATH = 'src/github.com/leonidlm/packer-builder-softlayer'
+
+script = <<SCRIPT
+SRCROOT="#{GOROOT}"
+
+# Install Go
+sudo apt-get update
+sudo apt-get install -y build-essential mercurial
+sudo hg clone -u release https://code.google.com/p/go $SRCROOT
+cd ${SRCROOT}/src
+sudo ./all.bash
+
+# Setup the GOPATH
+sudo mkdir -p #{GOPATH}
+cat <<EOF >/tmp/gopath.sh
+export GOPATH="#{GOPATH}"
+export PATH="#{GOROOT}/bin:\\$GOPATH/bin:\\$PATH"
+export SL_USERNAME=#{ENV['SL_USERNAME']}
+export SL_API_KEY=#{ENV['SL_API_KEY']}
+EOF
+sudo mv /tmp/gopath.sh /etc/profile.d/gopath.sh
+sudo chmod 0755 /etc/profile.d/gopath.sh
+
+# Install some other stuff we need
+sudo apt-get install -y curl git-core zip
+
+# Download and build Packer
+source /etc/profile.d/gopath.sh
+go get -u github.com/mitchellh/gox
+gox -build-toolchain
+go get -d -u github.com/mitchellh/packer
+cd $GOPATH/src/github.com/mitchellh/packer
+make updatedeps
+# These next two lines cause duplicate files in $GOPATH/bin
+make
+make dev
+
+# Build packer-builder-softlayer
+cd $GOPATH/#{PACKAGE_PATH}
+go build
+go install
+
+# Make sure the gopath is usable by vagrant
+sudo chown -R -f vagrant:vagrant $SRCROOT
+sudo chown -R -f vagrant:vagrant #{GOPATH} 
+
+echo "Ready for development. Begin with cd $GOPATH/#{PACKAGE_PATH}"
+
+SCRIPT
+
+Vagrant.configure(2) do |config|
+  config.vm.box = "chef/ubuntu-12.04"
+
+  config.vm.synced_folder '.', "#{GOPATH}/#{PACKAGE_PATH}", id: 'src' 
+
+  config.vm.provision 'shell', inline: script
+
+  config.vm.provider :virtualbox do |vb|
+    vb.memory = 2048
+    vb.cpus = 2
+    vb.gui = false
```

Comment:
Can you please parametrize all these values?
For example you can add the following lines at the top of the Vagrantfile:
```ruby
$memory = 2048
$cpus = 2
$gui = false
```

If you want to do something fancy you can fetch the values from ENV variables, but it isn't a must :)